plugins/inputs/ceph/ceph.go

```diff
@@ -85,9 +103,23 @@ func (c *Ceph) gatherAdminSocketStats(acc telegraf.Accumulator) error {
 			continue
 		}
 		for tag, metrics := range data {
-			acc.AddFields(measurement,
-				metrics,
-				map[string]string{"type": s.sockType, "id": s.sockID, "collection": tag})
+			for name, metric := range metrics {
+				var valueType telegraf.ValueType
+				var ok bool
+				if valueType, ok = c.SchemaMaps[*s][tag][name]; !ok {
+					acc.AddGauge(measurement,
+						map[string]interface{}{name: metric},
+						map[string]string{"type": s.sockType, "id": s.sockID, "collection": tag})
+				}
```

Comment:
I don't think this works. If e.g. `parseSchema` above fails, `c.SchemaMaps[*s]` is not valid and you likely cannot access `tag` nor `name`... Please also use implicit declarations as this creates more compact code
```suggestion
				valueType, ok = c.SchemaMaps[*s][tag][name]
				if !ok {
					acc.AddGauge(measurement,
						map[string]interface{}{name: metric},
						map[string]string{"type": s.sockType, "id": s.sockID, "collection": tag})
				}
```

Reply:
Reworked logic to have proper checks.